src/shared/utilities/childProcess.ts

```diff
@@ -104,6 +104,23 @@ export class ChildProcess {
         // [2] https://nodejs.org/api/child_process.html
         this.childProcess = crossSpawn.spawn(this.command, this.args, this.options)
 
+        function errorHandler(process: ChildProcess, params: ChildProcessStartArguments, error: Error): void {
+            process.processError = error
+            if (params.onError) {
+                params.onError(error)
+            }
+        }
+
+        // Emitted whenever:
+        //  1. Process could not be spawned, or
+        //  2. Process could not be killed, or
+        //  3. Sending a message to the child process failed.
+        // https://nodejs.org/api/child_process.html#child_process_class_childprocess
+        // We also register error event handlers on the output/error streams in case a lower level library fails
+        this.childProcess.on('error', (err) => { errorHandler(this, params, err) })
+        this.childProcess.stdout?.on('error', (err) => { errorHandler(this, params, err) })
```

Comment:
Does this potentially have knock on effects? Is there a scenario where it can succeed, but get error on stdout/stderr or the other way around?

Reply:
We treat a closed stream as end-of-process so it seems consistent to treat stream errors as a fatal error.